sdk/core/azure-core-amqp/samples/local_server_sample/local_server_sample.cpp

```diff
@@ -214,7 +214,7 @@ class SampleEvents : public ConnectionEvents,
 int main()
 {
   SampleEvents sampleEvents;
-  Network::_internal::SocketListener listener(5672, &sampleEvents);
+  Network::_internal::SocketListener listener(_detail::AmqpPort, &sampleEvents);
```

Comment:
Depending on who the audience of the sample is (end user vs sdk author), we want to avoid having the use of `_internal` types.

Regardless, samples shouldn't uses something from the  `_detail` namespace or include private headers. We can either consider making this constant `_internal` to be re-used by the SDK authors, or leave it as a magic number only within the sample.

Reply:
SDK authors won't ever use this constant. They will use URLs either provided by the customer, which may or may not have a port.

I've gone back to a magic constant in the sample.